src/main/webapp/app/text-editor/text-editor.component.html

```diff
@@ -70,20 +70,31 @@ <h2>
         </ng-template>
     </div>
 
-    <div class="row mt-4" *ngIf="result && result.completionDate && !hasComplaint">
+    <div class="row mt-4" *ngIf="result && result.completionDate && !hasComplaint && !hasRequestMoreFeedback">
         <button
             class="btn btn-primary"
             [class.not-allowed]="numberOfAllowedComplaints <= 0 || !isTimeOfComplaintValid"
-            (click)="showComplaintForm = !showComplaintForm"
+            (click)="toggleComplaintForm()"
             [disabled]="numberOfAllowedComplaints <= 0 || !isTimeOfComplaintValid"
             title="{{ numberOfAllowedComplaints <= 0 || !isTimeOfComplaintValid ? ('artemisApp.complaint.complaintNotAllowedTooltip' | translate) : '' }}"
         >
-            <!--TODO: use the non-temp message when the 'request more feedback' feature is implemented-->
-            {{ 'artemisApp.complaint.moreInfo_temp' | translate }}
+            {{ 'artemisApp.complaint.moreInfo' | translate }}
+        </button>
+        <button
+            class="btn btn-primary ml-1"
+            [class.not-allowed]="!isTimeOfComplaintValid"
+            (click)="toggleRequestMoreFeedbackForm()"
+            [disabled]="!isTimeOfComplaintValid"
+            title="{{ !isTimeOfComplaintValid ? ('artemisApp.moreFeedback.notAllowedTooltip' | translate) : '' }}"
+        >
+            {{ 'artemisApp.moreFeedback.button' | translate }}
         </button>
     </div>
 
     <div class="row" *ngIf="showComplaintForm || hasComplaint">
-        <jhi-complaint-form class="flex-grow-1" [resultId]="result.id" [allowedComplaints]="numberOfAllowedComplaints"></jhi-complaint-form>
+        <jhi-complaint-form class="flex-grow-1" [resultId]="result?.id" [allowedComplaints]="numberOfAllowedComplaints" (submit)="hasComplaint = true"></jhi-complaint-form>
```

Comment:
These two forms could also be wrapped together in their own component for better reuse.

Reply:
Done